src/core/border_router/routing_manager.cpp

```diff
@@ -817,6 +817,45 @@ bool RoutingManager::NetworkDataContainsUlaRoute(void) const
     return contains;
 }
 
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_REACHABLITY_CHECK_ICMP6_ERROR_ENABLE
+
+void RoutingManager::CheckReachbilityToSendIcmpError(const Message &aMessage, const Ip6::Header &aIp6Header)
+{
+    Ip6::MessageInfo messageInfo;
+
+    VerifyOrExit(IsRunning() && IsInitalPolicyEvaluationDone());
+
+    VerifyOrExit(!aIp6Header.GetDestination().IsMulticast());
+
+    // Validate that source matches the ULA OMR prefix (which is not
+    // infrastructure-derived).
+
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsEmpty());
+    VerifyOrExit(!mOmrPrefixManager.GetFavoredPrefix().IsInfrastructureDerived());
+    VerifyOrExit(aIp6Header.GetSource().MatchesPrefix(mOmrPrefixManager.GetFavoredPrefix().GetPrefix()));
```

Comment:
The spec text defines here any ULA OMR prefix with Preference = 11 (Low), not just the favored prefix.   And maybe the sending device had a different favored prefix than the BR at the time when it sent this packet?

Reply:
Thanks. I re-wrote these set of check to align it better with the spec text
It now validates that source matches a ULA OMR prefix with low preference (present in Network Data).